### ISSUE TITLE
Python 3: capture output broken up

### ISSUE THREAD

**Issue by pfuntner** [pfuntner/toys#55]
I had an instance of using `capture` where a line of stdout written to the screen was broken up:

```
Started: 2019-12-24 09:03:37.088830
2019-12-24 09:03:40.171839 +0:00:03.083009   3080 stdout 'Host       Hostname                       OS          IPs           Uptime '
2019-12-24 09:03:40.172746 +0:00:03.083916      0 stdout 'aws-rhel8  ip-172-31-58-182.ec2.internal  RedHat 8.0  52.87.227.97  0:55:00'
2019-12-24 09:03:41.173498 +0:00:04.084668   1000 stdout 'E'
2019-12-24 09:03:41.591093 +0:00:04.502263    417 stdout "xecuting: ['ansible-playbook', '/home/centos/tmpgyNRTk']"
```

It's almost as if the script thought that there was a interval between `E` and `xecuting`.  I'm not sure if any other output was mistreated this way but this enough to investigate.

I was using my `python3` branch, SHA 92c09c1a7444b407419dae68f98aedfdc840d1d5.

**Comment by pfuntner** [comment]
Yes, there are more examples:

```
2019-12-24 09:39:24.524261 +0:00:09.721171   1424 stdout 'TASK [foo : Update cache for the package manager] *********************************************************************************************************************************************************************************'
2019-12-24 09:39:24.943398 +0:00:10.140308    419 stdout 'ok: [aws-rhel7]'
2019-12-24 09:39:25.944605 +0:00:11.141515   1001 stdout 'o'
2019-12-24 09:39:38.800644 +0:00:23.997554  12856 stdout 'k: [gcp-rhel7]'
2019-12-24 09:39:39.801893 +0:00:24.998803   1001 stdout 'o'
2019-12-24 09:39:50.779654 +0:00:35.976564  10977 stdout 'k: [aws-rhel8]'
2019-12-24 09:39:51.780549 +0:00:36.977459   1000 stdout 'o'
2019-12-24 09:39:54.960628 +0:00:40.157538   3180 stdout 'k: [gcp-rhel8]'
2019-12-24 09:39:54.977279 +0:00:40.174189     16 stdout 'ok: [gcp-centos8]'
```

It's interesting that when this happens the first byte comes out on its own.  I haven't seen examples of more than one byte.

**Comment by pfuntner** [comment]
This example is really interesting:

```
2019-12-24 09:42:33.695370 +0:03:18.892280     95 stdout 'TASK [ansible-os_hardening : Ensure updates, patches are installed (yum Package Manager)] ***************************************************************************************************************************************************'
2019-12-24 09:42:33.741168 +0:03:18.938078     45 stdout 'skipping: [aws-rhel8]'
2019-12-24 09:42:33.812275 +0:03:19.009185     71 stdout 'skipping: [gcp-centos8]'
2019-12-24 09:42:34.813587 +0:03:20.010497   1001 stdout 's'
2019-12-24 09:43:28.822385 +0:04:14.019295  54008 stdout 'kipping: [gcp-rhel8]'
2019-12-24 09:43:29.824485 +0:04:15.021395   1002 stdout 'o'
2019-12-24 09:46:22.777520 +0:07:07.974430 172953 stdout 'k: [gcp-rhel7]'
2019-12-24 09:46:22.818489 +0:07:08.015399     40 stdout 'ok: [aws-rhel7]'
2019-12-24 09:46:22.872201 +0:07:08.069111     53 stdout 'TASK [ansible-os_hardening : Ensure updates, patches are installed (dnf Package Manager)] ***************************************************************************************************************************************************'
```

- At 09:43:29.824485, one byte was read
- The next byte wasn't read until nearly 3 minutes later!

Does `capture` have to turn off buffering on the `Popen` streams?

**Comment by pfuntner** [comment]
Fixed in 283fc3b7a8cb311415e466d4af361f2ac6970d1c
